### PR TITLE
Travis ci test

### PR DESCRIPTION
**Pull request by lightsprint09**
Tests are running on travis. Currently they all fail because it isn't possible to run a local node/socket.io server on travis while executing the tests.
My idea is to extract the server into another repo so it can be easily deployed to Heroku.

Do anyone have a better idea or an server to host the test server on?

Current Travis Build: https://travis-ci.org/lightsprint09/socket.io-client-swift

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+language: objective-c
```
Comment (nuclearace) [inline .travis.yml]: Is there no swift option yet?

**Comment by nuclearace** [comment]
I'll talk to @rauchg and see about what we can do for travis-ci testing. I know at one point it was turned on for this repo.

File: .travis.yml (hunk 1 above)
Comment (lightsprint09) [inline .travis.yml]: not not an explicit swift option.
But it works fine

**Comment by lightsprint09** [comment]
Ok. I think turning travis back shouldn't be a problem.

I think getting a test server working is harder. Any idea which is better than heroku

**Comment by lightsprint09** [comment]
This was not meant to be merged at this state. Tests are still failing on Travis and the test server problem is not solved yet. 

This breaks nothing. But if possible revert it. 

We need some kind of server to run the node/socket on to test the swift code against. As I mentioned before this could be a heroku server or something coustom. 

I want to hear your opinion about the server first before I will  implement it

**Comment by nuclearace** [comment]
Would it be possible to add a before_script that launches the node server? I know that the travis environment has node installed

**Comment by lightsprint09** [comment]
I don't think this is possible because Travis only provides a container runtime. This is of course iOS. I don't think it is possible to run code from a foreign container runtime

**Comment by nuclearace** [comment]
Well if that's the case, would it make sense to just stand up a test server that the tests point to?

**Comment by lightsprint09** [comment]
Yeah. Heroku is free but start up time is about 30 seconds. This slows tests down but should work. Another possibility is to host it somewhere els where the server stays always on

**Comment by nuclearace** [comment]
Yeah, I have a VPS that I use. The only thing is that if the tests change I'd have to manually update the test server. Or find a way to auto-update

**Comment by lightsprint09** [comment]
We could use github web hooks to automatically deploy new tests. Or something that works out of the box

**Comment by nuclearace** [comment]
I think temporarily I'll just manually update the server

**Comment by nuclearace** [comment]
So, it looks travis is running into some kind of problem

```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'API violation - multiple calls made to -[XCTestExpectation fulfill] for connect.'
```

**Comment by nuclearace** [comment]
Eck, it seems to be the same problem of sometimes the test pass and other times it fails

**Comment by nuclearace** [comment]
https://travis-ci.org/socketio/socket.io-client-swift/builds/78425251

**Comment by lightsprint09** [comment]
Do the tests pass locally ?

**Comment by nuclearace** [comment]
Yeah when run locally all the tests pass

**Comment by nuclearace** [comment]
The most common thing seems to be an error dealing with multiple calls to -[XCTestExpectation fulfill] like I posted above

**Comment by lightsprint09** [comment]
Is it normal that the debugger stops here
screenshot

**Comment by nuclearace** [comment]
If it's getting there that means either the client got released or something called stopPolling, which gets called in close.

**Comment by lightsprint09** [comment]
Is this an intended behavior?
client is not nil and reason is "canceled"

**Comment by nuclearace** [comment]
It looks like from your picture that something has called `close` on the engine. In which case it's normal to get to this check.

**Comment by nuclearace** [comment]
Although I believe that this check is from an earlier version of the client where the engine was only added once. I think it could probably be taken out

**Comment by lightsprint09** [comment]
I just found that travis does not support outside internet connection...
I´ll contact them for solution

**Comment by nuclearace** [comment]
I guess the _right_ thing to do would be to mock a socket.io server. But I really don't feel writing a mock server.

**Comment by lightsprint09** [comment]
year... but this would be a lot of work

**Comment by lightsprint09** [comment]
I was wrong.

Travis support said it is indeed possible to reach external services. I take another look into in

**Pull request merged**